prometheus/push/push.go

```diff
@@ -60,18 +60,18 @@ const contentTypeHeader = "Content-Type"
 // Note that all previously pushed metrics with the same job and other grouping
 // labels will be replaced with the metrics pushed by this call. (It uses HTTP
 // method 'PUT' to push to the Pushgateway.)
-func Registry(job string, grouping map[string]string, url string, reg prometheus.Gatherer) error {
-	return push(job, grouping, url, reg, "PUT")
+func FromGatherer(job string, grouping map[string]string, url string, g prometheus.Gatherer) error {
```

Comment:
Not entirely happy with the name, either. The concept is sound, IMHO, and in a certain way Go-specific (have interfaces for parts of the functionality of a concrete type).
All up for more intuitive naming, we just need ideas.